samples/bluetooth/periodic_adv_rsp/src/main.c

```diff
@@ -0,0 +1,388 @@
+
+#define NUM_RSP_SLOTS 5
+#define NUM_SUBEVENTS 5
+#define PACKET_SIZE   5
+#define NAME_LEN      30
+
+static K_SEM_DEFINE(sem_connected, 0, 1);
+static K_SEM_DEFINE(sem_discovered, 0, 1);
+static K_SEM_DEFINE(sem_written, 0, 1);
+static K_SEM_DEFINE(sem_disconnected, 0, 1);
+
+static struct bt_uuid_128 pawr_char_uuid =
+	BT_UUID_INIT_128(BT_UUID_128_ENCODE(0x12345678, 0x1234, 0x5678, 0x1234, 0x56789abcdef1));
+static uint16_t pawr_attr_handle;
+static const struct bt_le_per_adv_param per_adv_params = {
+	.interval_min = 0xFF,
+	.interval_max = 0xFF,
+	.options = 0,
+	.num_subevents = NUM_SUBEVENTS,
+	.subevent_interval = 0x30,
+	.response_slot_delay = 0x5,
+	.response_slot_spacing = 0x50,
+	.num_response_slots = NUM_RSP_SLOTS,
+};
+
+static struct bt_le_per_adv_subevent_data_params subevent_data_params[NUM_SUBEVENTS];
+static struct net_buf_simple bufs[NUM_SUBEVENTS];
+static uint8_t backing_store[PACKET_SIZE][NUM_SUBEVENTS];
+
+BUILD_ASSERT(ARRAY_SIZE(bufs) == ARRAY_SIZE(subevent_data_params));
+BUILD_ASSERT(ARRAY_SIZE(backing_store) == ARRAY_SIZE(subevent_data_params));
+
+static uint8_t counter;
+
+static void request_cb(struct bt_le_ext_adv *adv, const struct bt_le_per_adv_data_request *request)
+{
+	int err;
+	uint8_t to_send;
+	struct net_buf_simple *buf;
+
+	to_send = MIN(request->count, ARRAY_SIZE(subevent_data_params));
+
+	for (size_t i = 0; i < to_send; i++) {
+		buf = &bufs[i];
+		buf->data[buf->len - 1] = counter++;
+
+		subevent_data_params[i].subevent =
+			(request->start + i) % per_adv_params.num_subevents;
+		subevent_data_params[i].response_slot_start = 0;
+		subevent_data_params[i].response_slot_count = NUM_RSP_SLOTS;
+		subevent_data_params[i].data = buf;
+	}
+
+	err = bt_le_per_adv_set_subevent_data(adv, to_send, subevent_data_params);
+	if (err) {
+		printk("Failed to set subevent data (err %d)\n", err);
+	}
+}
+
+static bool print_ad_field(struct bt_data *data, void *user_data)
+{
+	ARG_UNUSED(user_data);
+
+	printk("    0x%02X: ", data->type);
+	for (size_t i = 0; i < data->data_len; i++) {
+		printk("%02X", data->data[i]);
+	}
+
+	printk("\n");
+
+	return true;
+}
+
+static struct bt_conn *default_conn;
+
+static void response_cb(struct bt_le_ext_adv *adv, struct bt_le_per_adv_response_info *info,
+		     struct net_buf_simple *buf)
+{
+	if (buf) {
+		printk("Response: subevent %d, slot %d\n", info->subevent, info->response_slot);
+		bt_data_parse(buf, print_ad_field, NULL);
+	} else {
+		printk("Failed to receive response: subevent %d, slot %d\n", info->subevent,
+		       info->response_slot);
+	}
+}
+
+static const struct bt_le_ext_adv_cb adv_cb = {
+	.pawr_data_request = request_cb,
+	.pawr_response = response_cb,
+};
+
+void connected(struct bt_conn *conn, uint8_t err)
+{
+	printk("Connected (err 0x%02X)\n", err);
+
+	if (err) {
+		bt_conn_unref(conn);
+
+		return;
+	}
+
+	default_conn = conn;
+}
+
+void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	if (conn != default_conn) {
```

Comment:
Would it make sense to enforce `CONFIG_BT_MAX_CONN=1` for this sample, and then not have to do this extra checking?